app/dts/layouts/common/75percent.dtsi

```diff
@@ -0,0 +1,291 @@
+            , <&key_physical_attrs 100 100 1500  200       0     0     0>
+            , <&key_physical_attrs 175 100    0  300       0     0     0>
+            , <&key_physical_attrs 100 100  175  300       0     0     0>
+            , <&key_physical_attrs 100 100  275  300       0     0     0>
+            , <&key_physical_attrs 100 100  375  300       0     0     0>
+            , <&key_physical_attrs 100 100  475  300       0     0     0>
+            , <&key_physical_attrs 100 100  575  300       0     0     0>
+            , <&key_physical_attrs 100 100  675  300       0     0     0>
+            , <&key_physical_attrs 100 100  775  300       0     0     0>
+            , <&key_physical_attrs 100 100  875  300       0     0     0>
+            , <&key_physical_attrs 100 100  975  300       0     0     0>
+            , <&key_physical_attrs 100 100 1075  300       0     0     0>
+            , <&key_physical_attrs 100 100 1175  300       0     0     0>
+            , <&key_physical_attrs 225 100 1275  300       0     0     0>
+            , <&key_physical_attrs 100 100 1500  300       0     0     0>
+            , <&key_physical_attrs 225 100    0  400       0     0     0>
+            , <&key_physical_attrs 100 100  225  400       0     0     0>
+            , <&key_physical_attrs 100 100  325  400       0     0     0>
+            , <&key_physical_attrs 100 100  425  400       0     0     0>
+            , <&key_physical_attrs 100 100  525  400       0     0     0>
+            , <&key_physical_attrs 100 100  625  400       0     0     0>
+            , <&key_physical_attrs 100 100  725  400       0     0     0>
+            , <&key_physical_attrs 100 100  825  400       0     0     0>
+            , <&key_physical_attrs 100 100  925  400       0     0     0>
+            , <&key_physical_attrs 100 100 1025  400       0     0     0>
+            , <&key_physical_attrs 100 100 1125  400       0     0     0>
+            , <&key_physical_attrs 175 100 1225  400       0     0     0>
+            , <&key_physical_attrs 100 100 1400  400       0     0     0>
+            , <&key_physical_attrs 100 100 1500  400       0     0     0>
+            , <&key_physical_attrs 125 100    0  500       0     0     0>
+            , <&key_physical_attrs 125 100  125  500       0     0     0>
+            , <&key_physical_attrs 125 100  250  500       0     0     0>
+            , <&key_physical_attrs 625 100  375  500       0     0     0>
+            , <&key_physical_attrs 100 100 1000  500       0     0     0>
+            , <&key_physical_attrs 100 100 1100  500       0     0     0>
+            , <&key_physical_attrs 100 100 1200  500       0     0     0>
+            , <&key_physical_attrs 100 100 1300  500       0     0     0>
+            , <&key_physical_attrs 100 100 1400  500       0     0     0>
+            , <&key_physical_attrs 100 100 1500  500       0     0     0>
+            ;
+    };
+    layout_75_iso: layout_75_iso {
```

Comment:
I think defining multiple layouts like this would cause all of the layouts in the included file to be compiled, even if only one of them is being used? IIRC Studio allows the user to select from all compiled layouts, in which case this would allow selecting layouts a keyboard might not support.

`/omit-if-no-ref/` could be used to avoid that, but then something would need to reference each layout that is supported somehow.

Would separating each layout into its own file be better? Alternatively, I think you could set `status = "disabled"` on every layout and make boards/shields explicitly set `status = "ok"` to enable the ones that are supported, but that's a bit awkward.

Reply:
I deal with this by using /delete-node/ for unsupported layouts. I'm not so sure about it though, maybe one file per layout is better

Reply:
Have pondered this for the day. There's a few things that complicate this:

Pros of "one file with related layouts":

* Position maps are really useful, especially if you have something like an ANSI with a split backspace, where if you lake a position map tweak, changing layouts will cause your "backspace" to disappear.
* For many designs, this would likely mean just one file to include.

Cons to "related layouts in one file":

* Risk of C&P because folks don't have good docs/references on how to only use the layouts they want.
* More "busy work" to enable/disable the layouts you want to use.
* Potentially *huge* shared layout files for some formats, e.g. 60%.
* Accidentally having variants enabled you don't need.

I *think* the benefits of users not having to figure out the position map bits would outweigh the other issues, if we can improve on the cons.

Perhaps:

* All layouts in the .dtsi files are `status = "disabled"` by default.
* Users will need to supply their own `transform` property for each layout the want to use *anyways* so the can also set `status = "okay"` at the same time. This matches the Zephyr convention for devices in SoCs, etc too.
* The physical position map code already handles ignoring mapping entries for layouts that are disabled. This means we can have a "complete" map for all possible variants that will only end up built with the pieces needed.
* We don't set a `kscan` or `transform` on any layouts in the layout .dtsi, and allow consumers to set them as appropriate.

Thoughts?

Reply:
Having rethought it and fiddled with testing things I'm tempted to suggest individual files per layout right now, we can always add a top level file per size (such as how behaviors.dtsi does it) down the line that incorporates the position mapping. Unless I'm mistaken if layouts are included but not provided a transform there's a compilation error, does that go away if they're disabled? I'm keen to make it as easy as possible for people downstream who just want an ansi 60 or hand wired 4x12.